### ISSUE TITLE
[java] AvoidInstantiatingObjectsInLoops: Instantiation and then Loop Exit

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#3832]
**Affects PMD Version:** 6.43.0

**Rule:** [AvoidInstantiationObjectsInLoops](https://pmd.github.io/pmd/pmd_rules_java_performance.html#avoidinstantiatingobjectsinloops)

**Description:**

`AvoidInstantiationObjectsInLoops` flags a problem with this code even though the instantiation can only be done one time.  One can replace the `throw caught;` with `break;` or `return;` and reproduce the problem.

**Code Sample demonstrating the issue:**

```
Throwable info;

while (true)
{
   try
   {
      // nothing to do
   }
   catch (Throwable caught)
   {
      info = new Throwable();   // PMD Violation: AvoidInstantiatingObjectsInLoops

      caught.addSuppressed(info);

      throw caught;
   }
}
```

**Expected outcome:**

PMD reports a violation at line `info = new Throwable();   // PMD Violation: AvoidInstantiatingObjectsInLoops`, but that's wrong. That's a false positive.

**Running PMD through:** Eclipse PMD Plugin
